src/bacnet/basic/object/bi.c

```diff
@@ -542,11 +699,12 @@ int Binary_Input_Read_Property(BACNET_READ_PROPERTY_DATA *rpdata)
 bool Binary_Input_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
 {
     bool status = false; /* return value */
-    int len = 0;
+    unsigned int object_index = 0;
+    BINARY_INPUT_DESCR *CurrentBI = NULL;
     BACNET_APPLICATION_DATA_VALUE value;
 
     /* decode the some of the request */
-    len = bacapp_decode_application_data(
+    int len = bacapp_decode_application_data(
```

Comment:
no need for ruling out  wp_data == NULL before proceding ? To avoid potential NULL pointer dereference.

Reply:
IT is guaranteed for the wp_data beying always ! NULL: https://github.com/se-apc/bacnet-stack/blob/ab442de61ae79558a340780c585028c16e4e1460/src/bacnet/basic/service/h_wp.c#L69 .